confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/DeleteTimeSeriesProcedure.java

```diff
@@ -0,0 +1,611 @@
+                .computeIfAbsent(dataNodeLocation, k -> new ArrayList<>())
+                .add(regionReplicaSet.getRegionId());
+          }
+        });
+    return dataNodeConsensusGroupIdMap;
+  }
+
+  @Override
+  protected void rollbackState(
+      ConfigNodeProcedureEnv env, DeleteTimeSeriesState deleteTimeSeriesState)
+      throws IOException, InterruptedException, ProcedureException {
+    RegionTask<TSStatus> rollbackStateTask =
+        new RegionTask<TSStatus>(
+            "roll back schema black list",
+            env,
+            env.getConfigManager().getRelatedSchemaRegionGroup(patternTree)) {
+          @Override
+          Map<Integer, TSStatus> sendRequest(
+              TDataNodeLocation dataNodeLocation, List<TConsensusGroupId> consensusGroupIdList) {
+            Map<Integer, TDataNodeLocation> dataNodeLocationMap = new HashMap<>();
+            dataNodeLocationMap.put(dataNodeLocation.getDataNodeId(), dataNodeLocation);
+            RollbackSchemaBlackListHandler handler =
+                new RollbackSchemaBlackListHandler(dataNodeLocationMap);
+            AsyncDataNodeClientPool.getInstance()
+                .sendAsyncRequestToDataNodeWithRetry(
+                    new TRollbackSchemaBlackListReq(consensusGroupIdList, patternTreeBytes),
+                    handler);
+            return handler.getDataNodeResponseStatusMap();
+          }
+        };
+    rollbackStateTask.execute();
+  }
+
+  @Override
+  protected boolean isRollbackSupported(DeleteTimeSeriesState deleteTimeSeriesState) {
+    return true;
+  }
+
+  @Override
+  protected DeleteTimeSeriesState getState(int stateId) {
+    return DeleteTimeSeriesState.values()[stateId];
+  }
+
+  @Override
+  protected int getStateId(DeleteTimeSeriesState deleteTimeSeriesState) {
+    return deleteTimeSeriesState.ordinal();
+  }
+
+  @Override
+  protected DeleteTimeSeriesState getInitialState() {
+    return DeleteTimeSeriesState.CONSTRUCT_BLACK_LIST;
+  }
+
+  public PathPatternTree getPatternTree() {
+    return patternTree;
+  }
+
+  public void setPatternTree(PathPatternTree patternTree) {
+    this.patternTree = patternTree;
+    requestMessage = patternTree.getAllPathPatterns().toString();
+    patternTreeBytes = preparePatternTreeBytesData(patternTree);
+  }
+
+  private ByteBuffer preparePatternTreeBytesData(PathPatternTree patternTree) {
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    DataOutputStream dataOutputStream = new DataOutputStream(byteArrayOutputStream);
+    try {
+      patternTree.serialize(dataOutputStream);
+    } catch (IOException ignored) {
+
+    }
+    return ByteBuffer.wrap(byteArrayOutputStream.toByteArray());
+  }
+
+  @Override
+  public void serialize(DataOutputStream stream) throws IOException {
+    stream.writeInt(ProcedureFactory.ProcedureType.DELETE_TIMESERIES_PROCEDURE.ordinal());
+    super.serialize(stream);
+    patternTree.serialize(stream);
```

Comment:
need serialize and deserialize requestMessage, patternTreeBytes .  If the leader breaks down, other follower nodes need to replay and execute the procedure after receiving the procedure. The attributes required by these states may have been initialized.

Reply:
These two attributes is generated from patternTree for state usage, which avoids redundant generation